src/Euclidean.jl

```diff
@@ -0,0 +1,43 @@
+@doc doc"""
+    Euclidean{T<:Tuple} <: Manifold
+
+Euclidean vector space $\mathbb R^n$.
+
+# Constructor
+
+    Euclidean(n)
+
+generates the $n$-dimensional vector space $\mathbb R^n$.
+
+   Euclidean(m, n)
+
+generates the $mn$-dimensional vector space $\mathbb R^{m \times n}$, whose
+elements are interpreted as $m \times n$ matrices.
+"""
+struct Euclidean{T<:Tuple} <: Manifold where {T} end
+
+Euclidean(n::Int) = Euclidean{Tuple{n}}()
+Euclidean(m::Int, n::Int) = Euclidean{Tuple{m,n}}()
+
+@generated manifold_dimension(::Euclidean{T}) where {T} = *(T.parameters...)
+
+struct EuclideanMetric <: RiemannianMetric end
+
+function local_metric(::MetricManifold{<:Manifold,EuclideanMetric}, x)
+    return Diagonal(ones(eltype(x), length(x)))
+end
+
+function inverse_local_metric(M::MetricManifold{<:Manifold,EuclideanMetric}, x)
+    return local_metric(M, x)
+end
+
+det_local_metric(M::MetricManifold{<:Manifold,EuclideanMetric}, x) = one(eltype(x))
+
+inner(::Euclidean, x, v, w) = dot(v, w)
+inner(M::MetricManifold{<:Manifold,EuclideanMetric}, args...) = inner(manifold(M), args...)
```

Comment:
Can you elaborate on the use case for this line? Because if I just would do `MetricManifold(SymmetricPositiveDefinite,EuclideanMetric)`(which is inconsistent in itself, yes!) this would return to the non-Euclidean metric the SymmetricPositiveDefinite matrices might have by default (the affine invariant). Yes that's not the only one, but `Hyperbolic(3)` could be used for the same inconsistency.

Reply:
Good point, this should forward to `dot` instead.